### DISCUSSION TITLE
Difference between next/router and next/navigation

### DISCUSSION THREAD

**Discussion by himself65** [vercel/next.js#48426] [Help] [answered]
### Summary

I cannot find a document about the detail of `next/navigation.`

**Comment by himself65** [comment]
https://beta.nextjs.org/docs/api-reference/use-router

**Comment by Wize6669** [comment, marked as answer]
In Next.js, next/router and next/navigation are both packages that provide routing capabilities for your application. However, they have different features and use cases.

next/router is the older routing package, and it is still supported by Next.js. It is a more lightweight package that is designed for basic routing needs. next/router is a good choice for simple applications or for applications that do not need the advanced features of next/navigation.

next/navigation is the newer routing package, and it is the recommended package for most Next.js applications. It is a more powerful package that provides a number of advanced features, such as:

App Router: The App Router is a new routing system that is designed for modern web applications. It provides a number of features, such as soft navigation, automatic data fetching, and pre-rendering.
Link component: The Link component is a new way to create links in Next.js applications. It is more performant than the traditional <a> tag, and it supports a number of features, such as prefetching and scroll restoration.
useRouter hook: The useRouter hook is a new hook that provides access to the router object. It is a more powerful hook than the traditional useRouter hook from next/router, and it provides access to a number of new features.
In general, you should use next/navigation for most Next.js applications. However, if you have a simple application or an application that does not need the advanced features of next/navigation, then you can still use next/router.

**Reply by andreconghau** [reply]
Nice explain.
Thanks!

**Reply by gurlivleenkainth2000** [reply]
Thanks @Wize6669 for this answer.

**Reply by Sanskarjain04** [reply]
Thanks @Wize6669
